### PR TITLE
#28 validate molfile is in V3000 format

### LINKED ISSUE #28: V3000 molfile validation
**As a** researcher
**I want** the API to ensure my V3000 molfile is properly formatted
**So that** when molfiles stored by the system are returned to me, I can be confident they are correct

**Request Example**
`POST https://api.chemreg.epa.gov/defined-compounds/`

**Acceptance Criteria:**
When I post a v3000 molfile to the API, It verifies that the molfile conforms to proper V3000 format prior to returning success
If the file is not v3000, the input is invalid (415 or 400?) (Message: "MolFile format is invalid.  Molfile v3000 format expected.").

**Test:**

### PR DESCRIPTION
**Pull request by debboutr**
closes #28 

I used [this](https://www.daylight.com/meetings/mug05/Kappler/ctfile.pdf) document to try to figure out the best way to pin the version of the molfile.

 There is a comment I made in the issue that's worth considering. 

Also, I was unable to find a way to get a V2000 faker out of the `defined_compound_factory` so I just duplicated the class, which I hope isn't necessary.

### CODE REVIEW

File: chemreg/compound/tests/factories.py [hunk 1]
```diff
@@ -20,6 +20,15 @@ class Meta:
         model = DefinedCompoundSerializer
 
 
+class DefinedCompoundFactory_V2000(DjangoSerializerFactory):
```
Comment (debboutr) [inline chemreg/compound/tests/factories.py:23]: is there a way that I don't have to do this to get a `v2000` molfile?

Comment (mcovalt) [reply chemreg/compound/tests/factories.py:23]: Allow passing `v2000=True` to the factory by either using:
* [a combination of `Param` and `LazyAttribute`](https://factoryboy.readthedocs.io/en/latest/reference.html#simple-parameters) **_or_**
* [`Trait`s](https://factoryboy.readthedocs.io/en/latest/reference.html#traits)

We'll want to keep v3000 as the default as this is closest to the "real" data.

**Comment by mwfrost** [comment]
Regarding your comment on the ticket...
https://app.zenhub.com/workspaces/chemreg-main-5df50d7e6348817dc38b2f05/issues/chemical-curation/chemcurator_django/28#issuecomment-613204667

I think the way the error messages flow back to the user is fine as it is. There may be edge cases that cause us to reorganize our approach, but we have the right idea.

**Comment by mwfrost** [comment]
The faker is excellent. Using that huge list of SMILES compounds is proving useful.
I tested the tests by changing the serializer so that `indigo.setOption("molfile-saving-mode", "3000")` was `indigo.setOption("molfile-saving-mode", "2000")` instead. 

Then when I ran `test_defined_compound_from_smiles()`, it used the SMILES string to create a structure in Indigo, but when it loaded that structure into the serializer, the serializer's `molfile_v3000` attribute looked like this:

```
<BoundField value=
  -INDIGO-04152015462D

 13 13  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  2  0  0  0  0
  2  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  2  0  0  0  0
  6  7  1  0  0  0  0
  7  8  2  0  0  0  0
  8  9  1  0  0  0  0
  9 10  2  0  0  0  0
 10  5  1  0  0  0  0
  6 11  1  0  0  0  0
 11 12  1  0  0  0  0
 11 13  2  0  0  0  0
M  END
 errors=[ErrorDetail(string='Structure is not in V3000 format.', code='invalid')]>
```

That error message is exactly what we want.

**Pull request merged**